_assets/javascripts/lib/distance-sorter.js

```diff
@@ -0,0 +1,167 @@
+/* eslist no-plusplus: 0 */
+/* global CRDS */
+/* jshint esversion: 6 */
+
+window.CRDS = window.CRDS || {};
+
+CRDS.DistanceSorter = class DistanceSorter {
+  constructor() {
+    this.locationDistances = [];
+    this.searchForm = undefined;
+    this.searchInput = undefined;
+    this.formSubmitButton = undefined;
+    this.cards = undefined;
+    this.locationFinder = undefined;
+    this.init();
+  }
+
+  init() {
+    this.searchForm = document.getElementById('locations-address-input');
+    this.searchForm.addEventListener('submit', this.handleFormSubmit.bind(this));
+    this.searchInput = this.searchForm.getElementsByTagName('input')[0];
+    this.formSubmit = this.searchForm.getElementsByTagName('button')[0];
+    this.formSubmit.disabled = false;
+    this.locationsCarousel = DistanceSorter.getLocationsCarousel();
+    this.cards = this.locationsCarousel.cards;
+    this.locationFinder = new CRDS.LocationFinder();
+  }
+
+  static getLocationsCarousel() {
+    return Object.values(CRDS._instances).find(
+      (instance) => DistanceSorter._isSectionLocationsCarousel(instance)
+    );
+  }
+
+  static _isSectionLocationsCarousel(instance) {
+    return instance.carousel !== undefined && instance.carousel.id === 'section-locations';
+  }
+
+  handleFormSubmit(event) {
+    this._disableForm();
+    this.getDistance()
+      .done((locationDistances) => { this._updateDomWithSortedCards(locationDistances); })
+      .fail((xhr, ajaxOptions, thrownError) => {
+        console.log(thrownError);
+        this.showError();
+        this.formSubmit.disabled = false;
+        this.removeLabels();
+      });
+  }
+
+  _disableForm() {
+    event.preventDefault();
+    this.formSubmit.disabled = true;
+  }
+
+  getDistance() {
+    this.locationDistances = [];
+    return this.locationFinder.getLocationDistances(this.searchInput.value);
+  }
+
+  _updateDomWithSortedCards(locationDistances) {
+    this._updateLocationDistancesArray(locationDistances);
+    this._display_sorted_locations_carousel();
+    this._resetFormOnSuccess();
+  }
+
+  _updateLocationDistancesArray(locationDistances) {
+    locationDistances.forEach(
+      locationObject => this._addToLocationDistancesArray(locationObject)
+    );
+  }
+
+  _addToLocationDistancesArray(locationObject) {
+    const locationDistance = this._createLocationDistanceObject(locationObject);
+    this.locationDistances.push(locationDistance);
+  }
+
+  _createLocationDistanceObject(locationObject) {
+    return {
+      location: locationObject.location.location,
+      distance: locationObject.distance
+    };
+  }
+
+  _display_sorted_locations_carousel() {
+    this.createDataAttributes();
+    this.appendDistances();
+    this.locationsCarousel.sortBy('distance');
+//  A anywhere card needs to be created in Contenful before
```

Comment:
I actually hard coded this card into the `HTML` - not sure if that was a good approach but it didn't really feel like Anywhere was a true location.